### PR TITLE
Use betydb polygons to get plot boundary definition

### PR DESCRIPTION
**Pull request by ZongyangLi**

### CODE REVIEW

File: plant_height/full_day_to_histogram.py [hunk 1]
```diff
@@ -331,39 +290,173 @@ def full_day_array_to_xlsx_for_roman(in_dir):
...
+    
+    
+    csv.close()
+    #submitToBety(out_file)
+    betydb.submit_traits(out_file, filetype='csv', betykey=betydb.get_bety_key(), betyurl=betydb.get_bety_url())
+    
+    
+    return
+
+def submitToBety(csvfile):
```
Comment (dlebauer) [inline plant_height/full_day_to_histogram.py]: how is this different from betydb.submit_traits used in line 338 above?

File: plant_height/full_day_to_histogram.py [hunk 2]
```diff
@@ -331,39 +290,173 @@ def full_day_array_to_xlsx_for_roman(in_dir):
...
+    betyAPI = "https://terraref.ncsa.illinois.edu/bety/api/beta/traits.csv"
+    betyKey = "D49SRRPIPFhJIiJ9XOlACRlc0BHhO3kzUJrnUBS2"
```
Comment (dlebauer) [inline plant_height/full_day_to_histogram.py]: @ZongyangLi I'm not sure how terrautils handles the api key, but it is not secure to post it in public code ... I've reset your API key (you can find the new one here: https://terraref.ncsa.illinois.edu/bety/users)

File: plant_height/full_day_to_histogram.py [hunk 3]
```diff
@@ -331,39 +290,173 @@ def full_day_array_to_xlsx_for_roman(in_dir):
...
 
-    xRange = 0
-    count = 0
+        if r.status_code == 200 or r.status_code == 201:
+            print("CSV successfully uploaded to BETYdb.")
+        else:
+            print("Error uploading CSV to BETYdb %s" % r.status_code)
+            print(r.text)
+
+def parse_site_from_plotNum_1728(plotNum):
```
Comment (dlebauer) [inline plant_height/full_day_to_histogram.py]: For next year we should discuss how to store plot information in the database if the current site names approach is insufficient (e.g. if useful we could add plot number as well as separate fields for col, row, subplot etc) in addition to the current human readable 'sitename' field.

File: plant_height/full_day_to_histogram.py (hunk 1 above)
Comment (ZongyangLi) [reply plant_height/full_day_to_histogram.py]: It's just my testing function. I can delete it now.

Comment (max-zilla) [reply plant_height/full_day_to_histogram.py]: I believe submitToBety has been replaced by betydb.submit_traits and this is just a leftover.

File: plant_height/full_day_to_histogram.py (hunk 2 above)
Comment (max-zilla) [reply plant_height/full_day_to_histogram.py]: @ZongyangLi what I've been doing is adding a --betykey parameter to pass in to the script where we can provide this value

Comment (ZongyangLi) [reply plant_height/full_day_to_histogram.py]: I failed to set betykey when I was doing the testing so I use a hard code here, sorry about that.

**Comment by max-zilla** [comment]
Very nice on such short notice, @ZongyangLi 

Do you have an example command line to run this script? I guess we would just move this out to Roger and...
```
parser.add_argument("-p", "--ply_dir", help="ply directory")
      parser.add_argument("-j", "--json_dir", help="json directory")
      parser.add_argument("-o", "--out_dir", help="output directory")
     parser.add_argument("-y", "--year", help="specify which year to process")
     parser.add_argument("-d", "--month", help="specify month")
     parser.add_argument("-s", "--start_date", help="specify start date")
     parser.add_argument("-e", "--end_date", help="specify end date")
```
So something like...
```
python full_day_to_histogram.py --mode all --ply_dir /projects/arpae/terraref/sites/ua-mac/Level_1/scanner3DTop --json_dir /projects/arpae/terraref/sites/ua-mac/raw_data/scanner3DTop --out_dir /projects/arpae/terraref/sites/ua-mac/Level_1/laser3d_plant_height
```
To process every day?

**Comment by ZongyangLi** [comment]
@max-zilla 

https://github.com/terraref/extractors-3dscanner/blob/plot_boundary_query/plant_height/laser_to_height.sh

Here is the example bash file. Since I am only able to start a 48 hours task on Roger, I would use '--mode date' and specify year, month, start date and end date. For each day it might need 4-8 hours processing

**Comment by max-zilla** [comment]
@ZongyangLi excellent. I will talk with @craig-willis and get this started by tomorrow morning.

**Pull request merged**